evmjit/libevmjit/BasicBlock.cpp

```diff
@@ -43,6 +44,7 @@ BasicBlock::LocalStack::LocalStack(BasicBlock& _owner) :
 
 void BasicBlock::LocalStack::push(llvm::Value* _value)
 {
+	assert(_value->getType() == Type::Word);
```

Comment:
That assert and some others protect LLVM IR validity. They help introducing changes in the compiler (detects places where the change should propagate to too).